service/tbc/rpc.go

```diff
@@ -640,6 +647,45 @@ func (s *Server) handleBlockInsertRawRequest(ctx context.Context, req *tbcapi.Bl
 	return &tbcapi.BlockInsertRawResponse{BlockHash: &hash}, nil
 }
 
+func (s *Server) handleKeystonesByHeightRequest(ctx context.Context, req *tbcapi.KeystonesByHeightRequest) (any, error) {
+	log.Tracef("handleKeystonesByHeightRequest")
+	defer log.Tracef("handleKeystonesByHeightRequest exit")
+
+	bhb, err := s.db.BlockHeaderBest(ctx)
+	if err != nil {
+		e := protocol.NewInternalError(err)
+		return &tbcapi.KeystonesByHeightResponse{
+			Error: e.ProtocolError(),
+		}, e
+	}
+
+	kssList, err := s.KeystonesByHeight(ctx, req.Height, req.Depth)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			return &tbcapi.KeystonesByHeightResponse{
+				BTCTipHeight: bhb.Height,
+				Error:        protocol.RequestErrorf("could not find keystones in range"),
+			}, nil
+		}
+		e := protocol.NewInternalError(err)
+		return &tbcapi.KeystonesByHeightResponse{
+			BTCTipHeight: bhb.Height,
+			Error:        e.ProtocolError(),
+		}, e
+	}
+
+	aks := make([]*hemi.L2KeystoneAbrev, 0, len(kssList))
+	for _, k := range kssList {
+		aks = append(aks,
```

Comment:
nit: I don't think that we need to `append` here if we already know the size (i.e. `len(kssList)`

Reply:
As the length of the slice is `0`, we do need `append` here - however, I think it would be more readable and cleaner if we changed this to:
```
aks := make([]*hemi.L2KeystoneAbrev, len(kssList))
for i, k := range kssList {
	aks[i] = // ...
```